services/checks/checks.go

```diff
@@ -493,16 +505,32 @@ func (s *Service) findTargets(serviceType models.ServiceType) ([]target, error)
 
 	for _, service := range services {
 		e := s.db.InTransaction(func(tx *reform.TX) error {
-			a, err := models.FindPMMAgentsForService(s.db.Querier, service.ServiceID)
+			agents, err := models.FindPMMAgentsForService(s.db.Querier, service.ServiceID)
 			if err != nil {
 				return err
 			}
-			if len(a) == 0 {
+			if len(agents) == 0 {
 				return errors.New("no available pmm agents")
 			}
-			agent := a[0]
 
-			dsn, err := models.FindDSNByServiceIDandPMMAgentID(s.db.Querier, service.ServiceID, a[0].AgentID, "")
+			var agent *models.Agent
+			for _, a := range agents {
+				agentVersion, err := version.Parse(*a.Version)
+				if err != nil {
+					return errors.Wrap(err, "failed to checks agent version")
+				}
+
+				if !s.pmmVersion.Less(agentVersion) {
```

Comment:
We should check for features (Action types) available in specific pmm-agent versions: MongoDBGetCmdLineOpts for 2.7.0+, other for 2.6.0+. We should not unconditionally skip all agents if they are older than PMM Server.

Reply:
Done